galaxy_ng/app/api/base.py

```diff
@@ -52,3 +59,57 @@ class GenericViewSet(LocalSettingsMixin, viewsets.GenericViewSet):
 
 class ModelViewSet(LocalSettingsMixin, viewsets.ModelViewSet):
     pass
+
+
+class RelatedFieldsBaseSerializer(serializers.Serializer):
+    """
+    Serializer only returns fields specified in 'include_related' query param.
+
+    This allows for fields that require more database queries to be optionally
+    included in API responses, which lowers the load on the backend. This is
+    intended as a way to include extra data in list views.
+
+    Usage:
+
+    This functions the same as DRF's base `serializers.Serializer` class with the
+    exception that it will only return fields specified in the `?include_related=`
+    query parameter.
+
+    Example:
+
+    MySerializer(RelatedFieldsBaseSerializer):
+        foo = CharField()
+        bar = CharField()
+
+    MySerializer will return:
+
+    {"foo": None} when called with `?include_related=foo` and {"foo": None, "bar" None}
+    when called with `?include_related=foo&include_related=bar`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        # This should only be included as a sub serializer and shouldn't be used for
+        # updating objects, so set read_only to true
+        kwargs['read_only'] = True
+        return super().__init__(*args, **kwargs)
+
+    def to_representation(self, instance):
+        result = OrderedDict()
```

Comment:
Does it need to be an `OrderedDict` ?

all dicts respects the insertion order on Python 3.